hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/BaseHoodieTableServiceClient.java

```diff
@@ -243,15 +245,17 @@ public Option<String> scheduleCompaction(Option<Map<String, String>> extraMetada
    * @param extraMetadata         Extra Metadata to be stored
    */
   public abstract void commitCompaction(String compactionInstantTime, HoodieCommitMetadata metadata,
-                                        Option<Map<String, String>> extraMetadata);
+                                        Option<Map<String, String>> extraMetadata,
+                                        Option<HoodieData<WriteStatus>> writeStatuses);
 
   /**
```

Comment:
Instead of `HoodieData<WriteStatus>`, can we use `List<WriteStatus>` ?

Reply:
And the compaction does not really change the recordKey -> location mappings, why we need the write status list as parameter then? Even for RLI, they should be ignored.

Reply:
Fair point. I just kept it for completeness. Even if we pass WriteStatus, the `getRecordIndexUpdate` method in metadata writer will be no-op (location change condition check should return early). I will remove the change for compaction.
> Instead of HoodieData<WriteStatus>, can we use List<WriteStatus> ?

The metadata writer works with `HoodieData`. If we change to List<WriteStatus> then it will trigger a collect as list. I think HoodieData makes more sense if further computation has to be performed.